test/functional/client_side_encryption_prose_tests.js

```diff
@@ -0,0 +1,974 @@
+'use strict';
+
+const mongodb = require('../../index');
+const chai = require('chai');
+const expect = chai.expect;
+chai.use(require('chai-subset'));
+
+// Tests for the ClientEncryption type are not included as part of the YAML tests.
+
+// In the prose tests LOCAL_MASTERKEY refers to the following base64:
+
+// .. code:: javascript
+
+//   Mng0NCt4ZHVUYUJCa1kxNkVyNUR1QURhZ2h2UzR2d2RrZzh0cFBwM3R6NmdWMDFBMUN3YkQ5aXRRMkhGRGdQV09wOGVNYUMxT2k3NjZKelhaQmRCZGJkTXVyZG9uSjFk
+
+describe(
+  'Client Side Encryption Prose Tests',
+  { requires: { clientSideEncryption: true } },
+  function() {
+    const dataDbName = 'db';
```

Comment:
I feel like a lot of this look similar between all these tests, is there a way to refactor it usefully?

Reply:
maybe once I merge both I can do a refactor PR to put lots of stuff (and stuff from APM) in common locations?